app/services/provider_authorisation.rb

```diff
@@ -9,19 +9,35 @@ def can_make_offer?(application_choice:, course_option_id: nil)
     return true if @actor.is_a?(SupportUser)
 
     supplied_course_option = CourseOption.find(course_option_id) if course_option_id
+
+    # enforce 'make_decisions' restriction
+    if supplied_course_option
+      training_provider = supplied_course_option.provider
+      ratifying_provider = supplied_course_option.course.accredited_provider
+    else
+      training_provider = application_choice.offered_option.provider
+      ratifying_provider = application_choice.offered_course.accredited_provider
+    end
+
+    related_providers = [training_provider, ratifying_provider].compact
+    return false if
+      FeatureFlag.active?('provider_make_decisions_restriction') &&
+        !actor_has_permission_to_make_decisions?(providers: related_providers)
+
+    # check (indirect) relationship between course_option and @actor
     if supplied_course_option && course_option_id != application_choice.course_option.id
-      application_choice_visible_to_user?(application_choice: application_choice) && \
+      application_choice_visible_to_user?(application_choice: application_choice) &&
         course_option_belongs_to_user_providers?(course_option: supplied_course_option)
     else
       application_choice_visible_to_user?(application_choice: application_choice)
     end
   end
 
   def can_change_offer?(application_choice:, course_option_id:)
-    new_course_option = CourseOption.find course_option_id
-    @actor.is_a?(SupportUser) || \
-      application_choice_visible_to_user?(application_choice: application_choice) && \
-        course_option_belongs_to_user_providers?(course_option: new_course_option)
+    can_make_offer?(
+      application_choice: application_choice,
+      course_option_id: course_option_id,
+    )
```

Comment:
Should we lose this method altogether and use `can_make_offer?` everywhere? Not 100% sure but it feels like it might be less to think about.